crates/lib/src/install.rs

```diff
@@ -93,9 +94,16 @@ where
     // Step 2: Create a XZ decoder
     let decoder = XzDecoder::new(stream);
 
-    // Step 3: Extract the tar archive
+    // Step 3: Make a temporary directory to extract the tarball into
+    let temp_dir = tempdir_in(config.pkgx_dir.join(&pkg.project))?;
+
+    // Step 4: Extract the tar archive
     let mut archive = Archive::new(decoder);
-    archive.unpack(&config.pkgx_dir).await?;
+    archive.unpack(&temp_dir).await?;
+
+    // Step 5: atomically move from temp dir to installation location
+    let partial_path = format!("{}/v{}", pkg.project, pkg.version);
+    fs::rename(temp_dir.path().join(&partial_path), &dst_path)?;
```

Comment:
Quick google suggests this won't work across mounts. So, probably wants a comment to find it when someone triggers it.

Reply:
It'll be the same mount, we use `tempdir_in` to make the tmp file in `$PKGX_DIR`.